### PR TITLE
Add tests for docs plots on GPU

### LINKED ISSUE #1200: [BUG][CV] example code is not tested on gpu
**Describe the bug**
Example code is tested only on cpu. If it was tested on gpu, we would have noticed that the line:
`result = suite.run(training_data, val_data, model, device)
`
In plot_detection_tutorial places the device in the wrong place / without a keyname.

**To Reproduce**
Run plot_detection_tutorial  on gpu

### PR DESCRIPTION
**Pull request by matanper**
close #1200

### CODE REVIEW

File: tests/vision/docs/test_plots.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from runpy import run_path
+import glob
+
+import torch
+
+
+def test_plots_on_gpu():
+    """If there is GPU available running all the docs plot files. Only makes sure the plots don't crash, and not \
+    testing any other display or functionality."""
+    if torch.cuda.is_available() or True:
```
Comment (ItayGabbay) [inline tests/vision/docs/test_plots.py:10]: or True?

File: tests/vision/docs/test_plots.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        plots_files = glob.glob('../docs/source/**/plot_*.py', recursive=True)
```
Comment (ItayGabbay) [inline tests/vision/docs/test_plots.py:11]: maybe better to work with Pathlib or os.path to support all OS

File: tests/vision/docs/test_plots.py (hunk 1 above)
Comment (matanper) [reply tests/vision/docs/test_plots.py:10]: oops, added for testing

File: tests/vision/docs/test_plots.py (hunk 2 above)
Comment (matanper) [reply tests/vision/docs/test_plots.py:11]: how do you know glob doesn't support all OS?

Comment (ItayGabbay) [reply tests/vision/docs/test_plots.py:11]: glob is fine. but the path you specified as parameter - better use os.path

Comment (matanper) [reply tests/vision/docs/test_plots.py]: I don't really understand why it's better, but updated

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**